FirebaseRemoteConfig/SwiftNew/ConfigRealtime.swift

```diff
@@ -0,0 +1,667 @@
+    backgroundChangeListener()
+  }
+
+  deinit {
+    dataTask?.cancel() // Ensure the task is cancelled when the object is deallocated
+    session?.invalidateAndCancel()
+  }
+
+  private static func setupHTTPRequest(_ options: FirebaseOptions,
+                                       _ namespace: String) -> URLRequest {
+    let url = Utils.constructServerURL(
+      domain: serverURLDomain,
+      apiKey: options.apiKey,
+      optionsID: options.gcmSenderID,
+      namespace: namespace
+    )
+    var request = URLRequest(url: url,
+                             cachePolicy: .reloadIgnoringLocalCacheData,
+                             timeoutInterval: timeoutSeconds)
+    request.httpMethod = httpMethodPost
+    request.setValue(applicationJSON, forHTTPHeaderField: contentTypeHeaderName)
+    request.setValue(applicationJSON, forHTTPHeaderField: acceptEncodingHeaderName)
+    request.setValue(gzip, forHTTPHeaderField: contentEncodingHeaderName)
+    request.setValue("true", forHTTPHeaderField: canRetry)
+    request.setValue(options.apiKey, forHTTPHeaderField: "X-Goog-Api-Key")
+    request.setValue(
+      Bundle.main.bundleIdentifier,
+      forHTTPHeaderField: iOSBundleIdentifierHeaderName
+    )
+    return request
+  }
+
+  private func setupSession() -> URLSession {
+    let config = URLSessionConfiguration.default
+    config.timeoutIntervalForResource = timeoutSeconds
+    config.timeoutIntervalForRequest = timeoutSeconds
+    return URLSession(configuration: config, delegate: self, delegateQueue: .main)
+  }
+
+  private func propagateErrors(_ error: Error) {
+    realtimeLockQueue.async { [weak self] in
+      guard let self else { return }
+      for listener in self.listeners {
+        if let listener = listener as? (RemoteConfigUpdate?, Error?) -> Void {
+          listener(nil, error)
+        }
+      }
+    }
+  }
+
+  // TESTING ONLY
+  @objc func triggerListenerForTesting(listener: @escaping (RemoteConfigUpdate?, Error?) -> Void) {
+    DispatchQueue.main.async {
+      listener(RemoteConfigUpdate(), nil)
+    }
+  }
+
+  // MARK: - HTTP Helpers
+
+  private func appName(fromFullyQualifiedNamespace fullyQualifiedNamespace: String) -> String {
+    return String(fullyQualifiedNamespace.split(separator: ":").last ?? "")
+  }
+
+  private func reportCompletion(onHandler completionHandler: (
+    (RemoteConfigFetchStatus, Error?) -> Void
+  )?,
+  withStatus status: RemoteConfigFetchStatus,
+  withError error: Error?) {
+    guard let completionHandler = completionHandler else { return }
+    realtimeLockQueue.async {
+      completionHandler(status, error)
+    }
+  }
+
+  private func refreshInstallationsToken(completionHandler: (
+    (RemoteConfigFetchStatus, Error?) -> Void
+  )?) {
+    guard let installations, !options.gcmSenderID.isEmpty else {
+      let errorDescription = "Failed to get GCMSenderID"
+      RCLog.error("I-RCN000074", errorDescription)
+      settings.isFetchInProgress = false
```

Comment:
This looks like a correct change, but just pointing out I noticed that the objc impl didn't do it.

Reply:
No action required.